12_TableHttp_Axios/readme.md

```diff
@@ -0,0 +1,151 @@
+# 12 Table Http with Axios
+
+Let's move forward with the table sample, this time we are going to replace the
+mock data by real one and we will make calls with the Axios package.
+
+We will take a startup point sample _11 TableMock_:
+
+## Summary steps:
+
+- Configure transpilation and add extra transpile step babel >> es5.
+- Update API in order to work with [Axios](https://github.com/axios/axios) and fetch data from Github API.
+- Update the _tableComponent_ in order to show this data.
+
+## Prerequisites
+
+Install [Node.js and npm](https://nodejs.org/en/) (v6.6.0 or newer) if they are not already installed on your computer.
+
+> Verify that you are running at least node v6.x.x and npm 3.x.x by running `node -v` and `npm -v` in a terminal/console window. Older versions may produce errors.
+
+## Steps to build it
+
+- Copy the content from _11 TableMock_ and execute:
+
+```
+npm install
+```
+
+- We install the Axios package:
+
+```
+  npm install --save axios
+```
+
+- Let's remove the file _memberMockData.ts_ in _src/api_ directory.
+
+- Let's replace _memberAPI_ load members with the fetch / promise one:
+
+_./src/api/memberAPI.ts_
+
+```javascript
+import { MemberEntity } from '../model/member';
+import Axios, { AxiosError, AxiosResponse } from 'axios';
+
+// Sync mock data API, inspired from:
+// https://gist.github.com/coryhouse/fd6232f95f9d601158e4
+class MemberAPI {
+  getAllMembers(): Promise<MemberEntity[]> {
+    const gitHubMembersUrl: string = 'https://api.github.com/orgs/lemoncode/members';
+
+    return Axios.get<MemberEntity[]>(gitHubMembersUrl)
+      .then(this.resolveMembers)
+      .catch(this.onError);
+  }
+
+  private onError(err: AxiosError): MemberEntity[] {
+    console.log(err.message);
+    return [];
+  }
+
+  private resolveMembers({ data }: AxiosResponse<MemberEntity[]>): MemberEntity[] {
+    const members: MemberEntity[] = data.map(gitHubMember => {
+      let member: MemberEntity = {
+        id: gitHubMember.id,
+        login: gitHubMember.login,
+        avatar_url: gitHubMember.avatar_url,
+      };
+
+      return member;
+    });
+
+    return members;
+  }
+}
+
+export const memberAPI = new MemberAPI();
```

Comment:
I thini its a good time to remove the class/singleton approach and use just a pure functional approach, what"s your take on this?

Reply:
A sample of this functional approach: https://github.com/Lemoncode/componentization-examples/blob/master/00_Hotels_List/01_implementation/src/api/hotel.ts

When we implemented the class/singleton approach was a mistake from our side

Reply:
Great! I agree with you. In my job I do it.
Perhaps should I add Axios as vendor in webpack config? what's your opinion about this?

Reply:
Good point, on thw other hand you can remove the return and the let variable

Reply:
Perfect. I hope finish in a minutes...